### DISCUSSION TITLE
React Native In-app Messaging - How to handle button clicks

### DISCUSSION THREAD

**Discussion by jamesawer3** [invertase/react-native-firebase#4568] [Q&A]
Hi, I'm using React Native In-app Messaging in my app and i have a doubt how to handle the button clicks i don't get any response when pressing the button.

**Comment by mikehardy** [comment]
There is no information in here on which to base an answer, sorry.
My best advice is to reach right into node_modules and hack in some console.log's so you can trace through what's happening. You can do the same in the underlying Java/Objective-C code (depending on your platform) and I recommend it so you understand the logic flow and can verify whatever arguments are going in to API calls and what return values are coming out

**Comment by jamesawer3** [comment]
Thanks for your reply @mikehardy I'm using [RNFirebase In-app Messaging](https://rnfirebase.io/in-app-messaging/usage). In the docs there is no handlers given.

![WhatsApp Image 2020-11-18 at 4 24 28 PM](https://user-images.githubusercontent.com/43437378/99548840-144d7b00-29df-11eb-9dca-c5140b0b988a.jpeg)

This is the UI designed using RNFirebase, in this when I press share button I have to handle that in my app code.

**Reply by mikehardy** [reply]
I think that's kind of the point of the in app messaging? It is perhaps the smallest module in this codebase, it does *almost* nothing except wrap the underlying APIs, which themselves do almost nothing. Which is another way of saying you can do whatever you like with it but you do have to implement that - there are no stock handlers

**Comment by jamesawer3** [comment]
Can I get some example's on how to handle the button click in my app code please.

**Reply by mikehardy** [reply]
We have no examples like that, sorry. I would advise looking at upstream docs from the google firebase SDKs - other than admob this is our least-used package (https://github.com/invertase/react-native-firebase/blob/master//README.md) so there is not a lot of shared experience to offer sorry.

**Comment by raienko** [comment]
You should use deeplinks.
Setup deeplink like `myapp://do_something` ( guide: https://reactnavigation.org/docs/deep-linking/ )
Then in firebase console add same action for the button.
When you press button, deeplinks listener will be triggered
